File: src/test/java/com/alibaba/support/junit/conditional/BelowJava7.java
package com.alibaba.support.junit.conditional;

import com.alibaba.support.junit.conditional.ConditionalIgnoreRule.IgnoreCondition;
import org.apache.commons.lang3.JavaVersion;
import org.apache.commons.lang3.SystemUtils;

/**
 * @see <a href="https://stackoverflow.com/a/23706899/922688">Getting Java version at runtime</a>
 */
public class BelowJava7 implements IgnoreCondition {
    public boolean isSatisfied() {
        return !SystemUtils.isJavaVersionAtLeast(JavaVersion.JAVA_1_7);
    }
}

File: library/src/test/java/com/alibaba/support/junit/conditional/BelowJava7.kt
package com.alibaba.support.junit.conditional

import com.alibaba.support.junit.conditional.ConditionalIgnoreRule.IgnoreCondition
import org.apache.commons.lang3.JavaVersion
import org.apache.commons.lang3.SystemUtils

/**
 * @see [Getting Java version at runtime](https://stackoverflow.com/a/23706899/922688)
 */
class BelowJava7 : IgnoreCondition {
    override fun isSatisfied(): Boolean =
            if (System.getProperty("java.specification.version").toDouble() >= 9) false
            else !SystemUtils.isJavaVersionAtLeast(JavaVersion.JAVA_1_7)
}
